### ISSUE TITLE
Baseline Data Split Module Implementation

### ISSUE THREAD

**Issue by cliu238** [cliu238/comsa_claude#3]
### Summary
Implement a simple, clean data split module for the baseline package that splits VA data by site column with support for multiple split strategies. This addresses the complexity issues in existing examples while providing the flexibility needed for future data analysis workflows.

### 🚀 Key Features
- **Simple, clean interface** - Much simpler than overcomplicated existing examples
- **Integration with baseline configuration** - Extends existing DataConfig system
- **Multiple split strategies** - train/test, cross-site validation, and stratified splits
- **Future analysis ready** - Designed for ML workflows and research
- **Comprehensive testing** - >90% test coverage requirement

#### Core Split Strategies
1. **Simple train/test split** - Standard sklearn train_test_split by site
2. **Cross-site validation** - Train on some sites, test on others for domain adaptation
3. **Stratified site split** - Maintain label distribution within each site

#### Technical Requirements
- Extend `DataConfig` with split-related configuration options
- Create `baseline/data/data_splitter.py` with clean `VADataSplitter` class
- Return DataFrames for flexibility (no complex file I/O)
- Follow CLAUDE.md principles: KISS, YAGNI, modularity
- Comprehensive unit tests in `tests/baseline/`
- Example usage script demonstrating all split modes

### 🔧 Implementation Details

**Configuration Extension:**
```python
# Add to DataConfig
split_strategy: Literal["train_test", "cross_site", "stratified_site"]
test_size: float = 0.3
random_state: int = 42
site_column: str = "site"
label_column: str = "va34"
train_sites: Optional[List[str]] = None
test_sites: Optional[List[str]] = None
```

**Core API:**
```python
from baseline.data.data_splitter import VADataSplitter

splitter = VADataSplitter(config)
splits = splitter.split_data(data)
# Returns: {"train": DataFrame, "test": DataFrame}
```

### 📚 Reference Materials
- `examples/cross_site_splitting.py` - Cross-site logic reference (simplify approach)
- `examples/dataset_splitting_v2.py` - Stratified splitting reference (remove complexity)
- `baseline/data/data_loader_preprocessor.py` - Architecture pattern to follow
- `baseline/config/data_config.py` - Configuration approach to extend

### ✅ Success Criteria
- [ ] Clean, simple API with single method
- [ ] Three split strategies implemented
- [ ] Integration with existing baseline configuration
- [ ] >90% test coverage
- [ ] Example usage script
- [ ] Documentation in README.md
- [ ] Follows CLAUDE.md principles

### 🎨 Design Principles
- **KISS**: Keep interface simple and focused
- **YAGNI**: Only implement what's needed
- **Modularity**: Clean integration with existing baseline module
- **Future-ready**: Designed for various data analysis workflows

This implementation will provide a clean, well-tested alternative to the overcomplicated existing examples while maintaining the flexibility needed for future VA data analysis.
